### ISSUE TITLE
Class-level @focus ignores testmatch

### ISSUE THREAD

**Issue by vbraun** [delfick/nose-focus#3]
Thanks for the nice plugin! I tripped over the following issue, though. The following testcase succeeds as expected, because the method name does not match the nose testmatch regex:
~~~~
class TestNotTestedMethod(TestCase):
    def foo_is_not_run_because_its_name_does_not_match(self):
        assert False
~~~~
But when I run it with `@focus` (obviously I want to do this where there are some testcases and some auxiliary methods), then non-tests are executed as well and the following fails:
~~~~
@focus
class TestNotTestMethodWithFocus(TestCase):
    def foo_is_not_run_because_its_name_does_not_match(self):
        assert False
~~~~
The expected behavior is probably that non-matching methods are not tested even with class-level `@focus`. This is with nose-focus 0.1.2 from pypi.

**Comment by delfick** [comment]
Hi, sorry I've taken so long to reply, it's been a very busy week! I should have time to have a look at this on the weekend, I'm sure it'll be easy :)

**Comment by delfick** [comment]
ok, that was a crazy week. Finally have time to look at this :)

**Comment by delfick** [comment]
Ok, that should fix it, and I've uploaded version 0.1.3
